### ISSUE TITLE
Support just-in-time loading of translations

### ISSUE THREAD

**Issue by swissspidy** [swissspidy/preferred-languages#22]
Current list preferred languages: de_CH -> de_DE -> en_US.
de_CH files for core are loaded, but for plugin `foo-bar` only a de_DE file exists.

Now, when WordPress calls `_load_textdomain_just_in_time()`, it ultimately calls `_get_path_to_translation_from_lang_dir()`. That function doesn't find the `foo-bar-de_DE.mo` file, because `get_locale()` points to de_CH (the first in the list above).

Everything is working fine when a plugin manually calls `load_plugin_textdomain()`, just not when only relying on the just-in-time loading.

**tl;dr**: I think we need a filter in `_get_path_to_translation_from_lang_dir()` or `_get_path_to_translation()` or `_load_textdomain_just_in_time`.
